### DISCUSSION TITLE
Persistent unaddressed UX issues: all empty lines erased from user prompts, expanding prompt box, nearly black scrollbar and corrupted text when rich text enabled

### DISCUSSION THREAD

**Discussion by rotemdan** [open-webui/open-webui#7302] [Issues]
# Bug Report

## Installation Method

Docker

## Environment

- **Open WebUI Version:** 0.4.4

- **Operating System:** Windows 11
- **Browser (if applicable):** Latest Brave

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

UI should, for the most part work normally. UI issues should be minor.

## Actual Behavior:

There are persistent UI issues that degrade the core experience, some are likely easy to fix.

## Description

**Bug Summary:**
I already reported several UI issues, that I perceive as important, as discussions, almost a month ago:
* [Paragraph breaks are being removed on initial and subsequent prompts (never seen before v0.3.35)](https://github.com/open-webui/open-webui/discussions/6539)
* [Empty prompt textbox is expanding on various circumstances (never seen before v0.3.35)](https://github.com/open-webui/open-webui/discussions/6494)
* [Scrollbar handle has become nearly totally black in v0.3.35](https://github.com/open-webui/open-webui/discussions/6472)
* [Prompt textbox is getting corrupted in an odd and inconsistent way (never seen before v0.3.35)](https://github.com/open-webui/open-webui/discussions/6477). **Note**: this one is the only one that is specific to "rich text" mode. All others occur both when rich text is enabled or disabled.

Seems like very few people have reported duplicates these or upvoted them, and it's possible the discussions got lost in a flood of many other ones.

The issues still persist in `v0.4.4`. I find it very odd that people don't find these issues more important. If I were developing a chat UI these would be high priority, since they relate to the core user experience, way before optional enhancements or more sophisticated features.

I would also likely have disabled "rich text" mode by default since it doesn't work correctly with code blocks, and overall, for users who aren't aware of it, gives the impression the application being "buggy" or "broken".

I think maybe people tend to get excited and motivated by the "new/cool stuff", or their own unique needs or edge cases, and less about "the obvious/core stuff", which they may assume the developer is 100% aware of, for whatever reason. Some people also have bad attention to details, and would simply experience the software as "buggy" but not really consciously work out what it is that causes that "feeling". There are also plenty of ultra-technical users using `open-webui`, that may not even notice UX issues (even obvious ones), or have very low standards of quality in regards to "presentation" and "aesthetics" (the kind that enjoy using command-lines and cryptic shell script because it makes them feel "cool").

## Reproduction Details

**Steps to Reproduce:**
Reproduction details are simple and are detailed in the individual issues.

## Note

I would like to express huge gratitude for making a great, free to use application, which I personally use every day. It seems that new features are being constantly added (many of which aren't really essential, used or needed by many), which is great, but I'm not sure about the effort that would be needed to maintain all of them in the long run, and the amount of resources that would remain to allocate to maintain and improve the core experience.

If you think that pull requests are the answer, remember also that the more complex the application becomes, the larger the learning curve for making contributions, especially casual ones. I'd also suggest that the core experience should be maintained by the core developer(s) (not casual pull requests), but the unessential extras possibly more by the larger community.

I assume you would convert this to a discussion (like most issues that are reported on the main tracker), but I still felt I had to post it here, because I'm not sure that otherwise they would get the attention they deserve.

**Comment by tjbck** [comment]
~https://github.com/open-webui/open-webui/discussions/6539: Cannot reproduce on my end with the latest.~ Resolved
~https://github.com/open-webui/open-webui/discussions/6494: We'll need a more reliable way to reproduce the issue.~ Resolved
https://github.com/open-webui/open-webui/discussions/6472: Intentional changes.
https://github.com/open-webui/open-webui/discussions/6477: Can reproduce, however, this is how [rich text input](https://claude.ai/new) works (which you have the option to toggle off); with that being said, rich text input is actively being improved so this might get resolved in the near future. 

Hope this helps, all of your issues posts would greatly benefit from following the issue reporting template, as I'm unable to fix the issue if I can't reproduce it.

**Reply by silentoplayz** [reply]
> https://github.com/open-webui/open-webui/discussions/6539: Cannot reproduce on my end with the latest.

I am able to reproduce #6539 on both Chrome and Firefox browsers. If I start a new chat with the query:
```
Hello








How are you doing?
```
without the triple backticks wrapping the text, then 4 empty lines will have been removed in between the text `Hello` and `How are you doing` when you edit your own response and count the empty lines still remaining. Note that the query provided above is just an example, and the number of lines that will be removed depends on the amount of empty lines between the text provided.

So the provided example results in:
```
Hello




How are you doing?
```
when editing the query.

___

> https://github.com/open-webui/open-webui/discussions/6494: We'll need a more reliable way to reproduce the issue.

Simply sending multi-line responses (2 lines of text or more in the prompt input) is enough to trigger #6494 on my end. This is also reproducible on my end in both Chrome and Firefox browsers.

![image](https://github.com/user-attachments/assets/ea694a43-bfd3-4811-974f-113abab4c806)

**Reply by tjbck** [reply]
Unable to reproduce the issue with the default rich text input, however I was able to reproduce with the textarea input. Taking a look! Appreciate the details!

**Reply by tjbck** [reply]
Fixed on dev, testing wanted here.

**Reply by silentoplayz** [reply]
> Fixed on dev, testing wanted here.

I can confirm both [#6539](https://github.com/open-webui/open-webui/discussions/6539) and [#6494](https://github.com/open-webui/open-webui/discussions/6494) issues are fixed on the dev branch.

**Comment by rotemdan** [comment]
Thanks! On `v0.4.6` I no longer see the expanding input box and erased line breaks issues. Feels like things are mostly back to normal again.

## Partial workaround for the dark scrollbar (on dark mode)

My current workaround to fix the dark scrollbar on Chromium browsers is using the [Stylus extension](https://chromewebstore.google.com/detail/stylus/clngdbkpkpeebahjckkjfobafhncgmne) and injecting the following CSS to the page:
```css
    ::-webkit-scrollbar-thumb {
        background-color: #b6b6b6 !important;
    }
```
Problem is, this workaround would impact any application running on `localhost`, including ones that have bright UI - meaning the scrollbar would become nearly white / invisible on those (basically shifting the problem to other local web apps).

## Rich-text mode issues

Although I disabled the rich-text mode a month ago, I enabled it back to see how things are progressing. The corrupted characters have improved but there are still some [major usability issues](https://github.com/open-webui/open-webui/issues/7444#), which I see as a strong reason to consider disabling it by default.

I opened the linked issue (#7444) today, about the rich-text issues, but it got immediately closed (not even converted to a discussion) and I got redirected to #6973 (basically an issue that tells people how to turn rich-text off), although the issues mentioned there weren't included anywhere else (as far as I know).

Here are the 4 screen captures from the closed issue (they demonstrate the problems in a visual way):

![Animation2](https://github.com/user-attachments/assets/19ebc6c1-37a0-4027-ac6a-395dcb9e51cb)
![Animation3](https://github.com/user-attachments/assets/814acfb4-97a0-4bd5-9e76-175c013601d0)
![Animation4](https://github.com/user-attachments/assets/25fac9c2-d2d5-473e-bbcf-7158048be886)
![Animation5](https://github.com/user-attachments/assets/eaf630d1-697f-4870-8aef-bb68a2c5cfc5)

These problems are very basic, and may affect anyone who writes more than the most basic form of messages (even doing basic things like copy-and-paste, or trying to write text at the beginning of a line that starts with styled text).

**Reply by michaelmarziani** [reply]
I searched through all the issues trying to develop an understanding of the origin of rich text in the project, as I was curious to read about possible applications of this feature. For me, I disabled it because when troubleshooting code, many stack traces pasted from the console would get completely garbled and the UI would often slow down trying to render it.

I couldn't find anything about why it was desired in the issues, so I'm still curious. Claude accepts rich text as well, but OpenAI, Gemini, Perplexity, and Mistral don't. If I bold a word, or put it in italics, or underline it, does that add emphasis in the mind of the AI?

**Comment by jdeepwell** [comment]
Also stumbled upon this. I tried to create a custom model for source code processing, but whenever I pasted some source code into the prompt, all the empty lines got eaten.

I love the rich text mode and am very sad to have to turn it off :((

I assumed that everything inside tripple single quotes would be preserved verbatim - which unfortunately it is not.

**Comment by R-Bower** [comment]
Frustration led me to a google search, which led me here.  I'd really love to see an improved Rich Text Input experience.  I've been spoiled by Slack.  They probably have the best implementation out there.
